kite-data/kite-data-core/src/main/java/org/kitesdk/data/View.java

```diff
@@ -128,4 +128,28 @@
    * @since 0.16.0
    */
   public URI getUri();
+
+  /**
+   * Returns an {@code Iterable} of non-overlapping {@link View} objects that
+   * partition the underlying {@link Dataset} and cover this {@code View}.
+   *
+   * The returned {@code View} objects are implementation-specific, but should
+   * represent reasonable partitions of the underlying {@code Dataset} based on
+   * its layout.
+   *
+   * The data contained by the union of each {@code View} in the
+   * {@code Iterable} must be a super-set of this {@code View}.
+   *
+   * Note that partitions are actual partitions under which data is stored.
+   * Implementations should omit any {@code View} that is empty.
+   *
+   * @return
+   *      An Iterable of the {@code View} that cover this {@code View}.
+   * @throws IllegalStateException
+   *      If the underlying {@code Dataset} is not partitioned.
+   * @throws UnsupportedOperationException
+   *      If the implementation does not support covering partitions.
+   * @since 1.1.0
+   */
+  public Iterable<PartitionView<E>> getCoveringPartitions();
```

Comment:
Returning the most granular level of partitions like you describe seems fine to me...it's predictable and easy to understand. I think we could do things in addition to this (and probably outside the scope of this change) to make it easier for management tooling. 

Just a thought, but some possible methods that could help do this on the PartitionView interface: 

* A getParentPartition() method returning a parent, if it exists. This would make it easier to "walk up" from the more granular views.
* A getPartitionKeys(), which returns a map of the keys defining a partition: ("year" -> "2015", "month" -> "5", "day" -> 22), which allows tooling to find the partition they want to work with.

This is a bit hypothetical...something like this might be best to add in a subsequent JIRA once we have a concrete design for arching data. This is assuming you don't mind adding new methods to the PartitionView interface after it has been published. (One of those technically non-passive but doubtful to break users interface additions.)